File: AEDS2/Tp1/q17/RPalindromo.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

//checa se a string é palindromo
bool isPalindromo ( char s[], int y ) {
        bool result = true;
        int l;
        l = strlen(s);
        if ( y < l/2 ) {
            if ( s[y] != s[l-y-1] ) {
                result = false;
            } else {
                result = isPalindromo(s,++y);
            }
        }
        return result;
    }

//checa se a string é "FIM"
bool isFim (char s[]) {
    bool result = false;
    if (s[0] == 'F' && s[1] == 'I' && s[2] == 'M') { result = 1; } 
    return result;
}

//main que le entrada e retorna se é palindromo
int main () {
    char s[1000];
    scanf( "%[^\n]", s );
    while ( ! isFim(s) ) {
        if (isPalindromo(s,0)) {
            printf("SIM\n");
        } else {
            printf("NAO\n");
        }
        scanf( " %[^\n]", s );
    }

    return 0;
}

File: AEDS2/Tp1/q16/RPalindromo.java
import java.util.*;

//checa se a string é palindromo
class RPalindromo {
    public static boolean isPalindromo ( String s, int y ) {
        boolean result = true;
        if ( y < s.length()/2 ) {
            if ( s.charAt(y) != s.charAt(s.length() - y - 1) ) {
                result = false;
            }
            else {
                y++;
                result = isPalindromo(s,y);
            }
        }
        return result;
    }

    //main que le entrada e retorna se é palindromo
    public static void main (String args[]) {
        //Scanner obj = new Scanner (System.in);
        String s;
        //s = obj.nextLine();
        s = MyIO.readLine();
        while ( ! (s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M') ) {
            if ( isPalindromo(s,0) ) {
                System.out.println("SIM");
            } else {
                System.out.println("NAO");
            }
            //s = obj.nextLine();
            s = MyIO.readLine();
        }
    }
}

//s.charAt(y).equals( s.charAt(s.length() - y - 1)